File: docs/examples/record_api_dotnet/List.cs
using TrailBase;
using System.Text.Json.Nodes;

public partial class Examples {
  public static async Task<ListResponse<JsonObject>> List(Client client) =>
    await client.Records("movies").List(
        pagination: new Pagination(limit: 3),
        order: ["rank"],
        filters: [
          // Multiple filters on same column: watch_time between 90 and 120 minutes
          new Filter(column:"watch_time", op:CompareOp.GreaterThanOrEqual, value:"90"),
          new Filter(column:"watch_time", op:CompareOp.LessThan, value:"120"),
          // Date range: movies released between 2020 and 2023
          new Filter(column:"release_date", op:CompareOp.GreaterThanOrEqual, value:"2020-01-01"),
          new Filter(column:"release_date", op:CompareOp.LessThanOrEqual, value:"2023-12-31"),
        ]);
}

File: docs/examples/record_api_ts/src/list.ts
import type { Client, ListResponse } from "trailbase";

export const list = async (client: Client): Promise<ListResponse<object>> =>
  await client.records("movies").list({
    pagination: {
      limit: 3,
    },
    order: ["rank"],
    filters: [
      // Multiple filters on same column: watch_time between 90 and 120 minutes
      {
        column: "watch_time",
        op: "greaterThanOrEqual",
        value: "90",
      },
      {
        column: "watch_time",
        op: "lessThan",
        value: "120",
      },
      // Date range: movies released between 2020 and 2023
      {
        column: "release_date",
        op: "greaterThanOrEqual",
        value: "2020-01-01",
      },
      {
        column: "release_date",
        op: "lessThanOrEqual",
        value: "2023-12-31",
      },
    ],
  });
